src/components/Game/Game.jsx

```diff
@@ -0,0 +1,161 @@
+import {History} from "../History/History";
+import '../../App.css'
+import {GameMenu} from "../GameMenu/GameMenu";
+import {GameInfo} from "../GameInfo/GameInfo";
+import {GAME_SYMBOLS} from "../../scripts/libraries";
+import {calculateBotMove} from "../../scripts/bot-move";
+import {calculateScore} from "../../scripts/calculate-score";
+import {initStartState} from "../../scripts/initStartState";
+import {getHistory, getSquares} from "../../scripts/helpers";
+
+class Game extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = initStartState();
+        this.jumpToHandleClick = this.jumpTo.bind(this);
+        this.boardHandleClick = this.handleClick.bind(this);
+        this.startNewGameHandler = this.startNewGame.bind(this);
+        this.enableBotModeHandler = this.enableBotMode.bind(this);
+        this.disableBotModeHandler = this.disableBotMode.bind(this)
+        this.resetScoreHandler = this.resetScore.bind(this);
+    }
+
+    componentDidMount() {
+        const data = getLocalStorageData(LocalStorageKeys.game);
+        if (data) {
+            this.setState(data);
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState) {
+        if (!this.state.isGameOver) {
+            const lastSquares = this.state.history[this.state.history.length - 1].squares;
+            const winner = calculateWinner(lastSquares);
+            if (winner) {
+                const {winsX, winsO} = this.state.score;
+                const {updatedWinsX, updatedWinsO} = calculateScore(winner, winsX, winsO);
+                this.setState({
+                    isGameOver: true,
+                    score: {
+                        winsX: updatedWinsX,
+                        winsO: updatedWinsO,
+                    }
+                })
+            } else if (lastSquares.indexOf(null) < 0) {
+                this.setState({
+                    isGameOver: true,
+                })
+            }
+        }
+        updateLocalStorageData(LocalStorageKeys.game, this.state);
+    }
+
+    fillSquare(i) {
+        const history = getHistory(this.state);
+        const currentSquares = getSquares(history);
+        if (!(calculateWinner(currentSquares) || currentSquares[i])) {
+            this.setState((prevState) => {
+                const prevHistory = getHistory(prevState);
+                const squares = [...getSquares(prevHistory)];
+                squares[i] = prevState.xIsNext ? GAME_SYMBOLS.X : GAME_SYMBOLS.O;
+                return {
+                    ...prevState,
+                    history: prevHistory.concat([{squares}]),
+                    xIsNext: !prevState.xIsNext,
+                    stepNumber: prevHistory.length,
+                }
+            });
+        }
+    }
+
+
+    handleClick(i) {
+        if (!(this.state.isMoveBlocked || this.state.isGameOver)) {
+            this.fillSquare(i);
+            const squares = this.state.history[this.state.stepNumber].squares;
```

Comment:
this.state.history[this.state.stepNumber].squares - the same

Reply:
```suggestion
            const squares = getSquares(this.state.history, this.state.stepNumber);
```
The code of method getSquares
`export function getSquares(history, length = history.length - 1) {
    return history[length].squares;
}`
Is it ok?